pkg/api/datasources.go

```diff
@@ -354,17 +355,20 @@ func validateJSONData(ctx context.Context, jsonData *simplejson.Json, cfg *setti
 	if list == nil {
 		return nil
 	}
-	teamHTTPHeadersJSON := datasources.TeamHTTPHeadersJSONData{}
-	if jsonData != nil {
-		jsonData, err := jsonData.MarshalJSON()
+	if !features.IsEnabled(featuremgmt.FlagTeamHttpHeaders) {
+		return nil
+	}
+	teamHTTPHeadersJSON := datasources.TeamHTTPHeaders{}
+	if jsonData != nil && jsonData.Get("teamHttpHeaders") != nil {
+		jsonData, err := jsonData.Get("teamHttpHeaders").MarshalJSON()
```

Comment:
I would suggest to extract JSON unmarshaling part from 
`func (ds DataSource) TeamHTTPHeaders() (TeamHTTPHeaders, error) {`
in models to share the same code.